pkg/spec/merge.go

```diff
@@ -338,79 +312,108 @@ func mergeProperties(properties, properties2 spec.SchemaProperties, path *field.
-			Default:             responses.Default, // default will be the same because we are generating it
-			StatusCodeResponses: make(map[int]spec.Response),
-		},
-	}
-
-	statusCodeResponses := responses.StatusCodeResponses
-	statusCodeResponses2 := responses2.StatusCodeResponses
+	retResponses := spec.NewResponses()
 
 	// go over first responses list
 	// 1. merge mutual response code responses
-	// 2. add non mutual response code responses
-	for code, response := range statusCodeResponses {
-		if response2, ok := statusCodeResponses2[code]; ok {
-			mergedResponse, conflicts := mergeResponse(response, response2, path.Child(strconv.Itoa(code)))
+	// 2. add non-mutual response code responses
+	for code, response := range responses {
+		if response2, ok := responses2[code]; ok {
+			mergedResponse, conflicts := mergeResponse(response.Value, response2.Value, path.Child(code))
 			retConflicts = append(retConflicts, conflicts...)
-			retResponses.StatusCodeResponses[code] = *mergedResponse
+			retResponses[code] = &spec.ResponseRef{Value: mergedResponse}
 		} else {
-			retResponses.StatusCodeResponses[code] = statusCodeResponses[code]
+			retResponses[code] = responses[code]
 		}
 	}
 
-	// add non mutual parameters from the second list
-	for code := range statusCodeResponses2 {
-		if _, ok := statusCodeResponses[code]; !ok {
-			retResponses.StatusCodeResponses[code] = statusCodeResponses2[code]
+	// add non-mutual parameters from the second list
+	for code := range responses2 {
+		if _, ok := responses[code]; !ok {
+			retResponses[code] = responses2[code]
 		}
 	}
 
 	return retResponses, retConflicts
 }
 
-func mergeResponse(response, response2 spec.Response, path *field.Path) (*spec.Response, []conflict) {
+func mergeResponse(response, response2 *spec.Response, path *field.Path) (*spec.Response, []conflict) {
 	var retConflicts []conflict
 	retResponse := spec.NewResponse()
+	if response.Description != nil {
+		retResponse.WithDescription(*response.Description)
+	} else if response2.Description != nil {
+		retResponse.WithDescription(*response2.Description)
+	}
 
-	schema, conflicts := mergeSchema(response.Schema, response2.Schema, path.Child("schema"))
-	retResponse.Schema = schema
+	content, conflicts := mergeContent(response.Content, response2.Content, path.Child("content"))
+	if len(content) > 0 {
+		retResponse.WithContent(content)
+	}
 	retConflicts = append(retConflicts, conflicts...)
 
 	headers, conflicts := mergeResponseHeader(response.Headers, response2.Headers, path.Child("headers"))
-	retResponse.Headers = headers
+	if len(headers) > 0 {
+		retResponse.Headers = headers
+	}
 	retConflicts = append(retConflicts, conflicts...)
 
 	return retResponse, retConflicts
 }
 
-func mergeResponseHeader(headers, headers2 map[string]spec.Header, path *field.Path) (map[string]spec.Header, []conflict) {
+func mergeContent(content spec.Content, content2 spec.Content, path *field.Path) (spec.Content, []conflict) {
```

Comment:
maybe better to add a unit test for this

Reply:
all use cases are being covered with `Test_mergeRequestBody`